<!-- 
  Example launch file for octomap_server mapping: 
  Listens to incoming PointCloud2 data and incrementally builds an octomap. 
  The data is sent out in different representations. 

  Copy this file into your workspace and adjust as needed, see
  www.ros.org/wiki/octomap_server for details  
-->
<launch>
	<arg name="cloud_in_trav" default="/region_growing_traversability_pc_map" />
	<arg name="cloud_in_obst_pos" default="/region_growing_obstacles_pos_pc_map" />
	<arg name="cloud_in_obst_neg" default="/region_growing_obstacles_neg_pc_map" />
	<arg name="frame" default="world" />

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server_traversable">
		<param name="resolution" value="0.01" />
		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<!-- <param name="frame_id" value="arco/base_link"/> -->
		<param name="frame_id" value="$(arg frame)"/>
		<param name="occupancy_min_z" value="-05"/>
		<param name="occupancy_max_z" value="10"/>
		<param name="pointcloud_min_z" value="-6"/>
		<param name="pointcloud_max_z" value="10"/>
		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model/max_range" value="200.0" />
		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="$(arg cloud_in_trav)" />
		<remap from="/octomap_full" to="/octomap_traversable" />
        <remap from="/octomap_point_cloud_centers" to="/octomap_point_cloud_traversable"/>
	</node>

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server_obstacles_pos">
		<param name="resolution" value="0.01" />
		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<!-- <param name="frame_id" value="arco/base_link"/> -->
		<param name="frame_id" value="$(arg frame)"/>
		<param name="occupancy_min_z" value="-05"/>
		<param name="occupancy_max_z" value="10"/>
		<param name="pointcloud_min_z" value="-6"/>
		<param name="pointcloud_max_z" value="10"/>
		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model/max_range" value="200.0" />
		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="$(arg cloud_in_obst_pos)" />
		<remap from="/octomap_full" to="/octomap_obstacles_positive" />
        <remap from="/octomap_point_cloud_centers" to="/octomap_point_cloud_obstacles_pos"/>
	</node>

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server_obstacles_neg">
		<param name="resolution" value="0.01" />
		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<!-- <param name="frame_id" value="arco/base_link"/> -->
		<param name="frame_id" value="$(arg frame)"/>
		<param name="occupancy_min_z" value="-05"/>
		<param name="occupancy_max_z" value="10"/>
		<param name="pointcloud_min_z" value="-6"/>
		<param name="pointcloud_max_z" value="10"/>
		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model/max_range" value="200.0" />
		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="$(arg cloud_in_obst_neg)" />
		<remap from="/octomap_full" to="/octomap_obstacles_negative" />
        <remap from="/octomap_point_cloud_centers" to="/octomap_point_cloud_obstacles_neg"/>
	</node>

</launch>